core/testcontainers/core/waiting_utils.py

```diff
@@ -78,7 +78,10 @@ def wait_for(condition: Callable[..., bool]) -> bool:
 
 
 def wait_for_logs(
-    container: "DockerContainer", predicate: Union[Callable, str], timeout: Optional[float] = None, interval: float = 1
+    container: "DockerContainer",
+    predicate: Union[Callable[..., Any], str],
```

Comment:
Shouldn't the callable here be `Callable[(str,) bool]`? It's used as `if predicate(stdout) or predicate(stderr):`

Reply:
in java it also takes regex i dont remember if that is the case here but i will leave this thread open to review when and if this PR goes on. I think it belongs in a draft state for now. i just wanted to record the results of going through the exercise to add types. for example, some of the places we return ports are actually `str | int` (well, union, for `3.8` compatibility)